services/spar/test-integration/Test/Spar/APISpec.hs

```diff
@@ -712,7 +714,11 @@ specCRUDIdentityProvider = do
           metadata <- makeTestIdPMetadata
           idp <- call $ callIdpCreate (env ^. teSpar) (Just owner) metadata
           idp' <- call $ callIdpGet (env ^. teSpar) (Just owner) (idp ^. idpId)
-          liftIO $ idp `shouldBe` idp'
+          rawmeta <- runSparCass $ Data.getIdPRawMetadata (idp ^. idpId)
```

Comment:
Shouldn't you call the API here instead of the database? The database is an implementation detail considering this is the `APISpec`

Reply:
Or don't we have an API call to retrieve  `IdPMetadataValue`

Reply:
nope, we don't, and that's intentional.  we need this for trouble-shooting production issues, and if we do we have access to the database directly, so the database schema is not an implementation detail here, but the interface.  :)

Reply:
(there are other examples in the integration tests where we (usually me) talk to the database directly.  this is usually to make sure the internal state is inconsistent even when it is not exposed directly, and tricking the interface into giving indirect evidence of this consistency is anywhere between awkward and infeasible.

i'm happy to have a fight over this over beer.  :)

Reply:
I'm convinced. Just think it will be useful to have an endpoint in the future.  CS will not be poking into the cassandra database manually when debugging issues with customers. Given that _we_ are currently the customer support for SSO it is fine, but this will be a blocker in the future when we delegate this to others.

Reply:
We fought over coffee instead.